index.js

```diff
@@ -7,13 +7,26 @@ const YAML = require('js-yaml')
 const TOML = { parse: require('toml-j0.4').parse, stringify: require('tomlify-j0.4') }
 const INI = require('ini')
 
+/**
+ * Removes '.' in file extensions
+ * @param  {string} type input file extension
+ * @return {string}      file extension without '.'
+ */
+const validType = (type) => {
+    if (type.charAt(0) === '.') {
+        type = type.substr(1)
+    }
+    return type
+}
+
 const parse = (string, type) => {
     if (!string) throw new Error('missing string')
     if (!type) throw new Error('missing type')
+    type = validType(type)
     switch (type) {
         case 'json': return JSON.parse(string)
         case 'json5': return JSON5.parse(string)
-        case 'hjson': return HJSON.parse(string)
+        case 'hjson': return HJSON.parse(string.toString())
```

Comment:
Can you explain why the `string.toString()` is necessary, please? Seems kind of odd to me.